Example/Messaging/Tests/FIRMessagingPubSubTest.m

```diff
@@ -87,4 +104,30 @@ - (void)testRemoveTopicPrefix {
   XCTAssertEqualObjects(topic, kTopicName);
 }
 
+-(void)testTopicListArchive {
+  MockPendingTopicsListDelegate *notReadyDelegate = [[MockPendingTopicsListDelegate alloc] init];
+  notReadyDelegate.isReady = NO;
+  FIRMessagingPendingTopicsList *topicList = [[FIRMessagingPendingTopicsList alloc] init];
+  topicList.delegate = notReadyDelegate;
+
+  // There should be 3 batches as actions are different than the last ones.
+  [topicList addOperationForTopic:@"/topics/0" withAction:FIRMessagingTopicActionSubscribe completion:nil];
+  [topicList addOperationForTopic:@"/topics/1" withAction:FIRMessagingTopicActionUnsubscribe completion:nil];
+  [topicList addOperationForTopic:@"/topics/2" withAction:FIRMessagingTopicActionSubscribe completion:nil];
+  XCTAssertEqual(topicList.numberOfBatches, 3);
+
+  id mockClientDelegate = OCMStrictProtocolMock(@protocol(FIRMessagingClientDelegate));
+  id mockReachability = OCMClassMock([GULReachabilityChecker class]);
+  id mockRmqManager = OCMClassMock([FIRMessagingRmqManager class]);
+  FIRMessagingClient *client = [[FIRMessagingClient alloc]
+                                initWithDelegate:mockClientDelegate
+                                reachability:mockReachability
+                                rmq2Manager:mockRmqManager];
+  FIRMessagingPubSub *pubSub = [[FIRMessagingPubSub alloc] initWithClient:client];
+  [pubSub archivePendingTopicsList:topicList];
+ 
+  [pubSub restorePendingTopicsList];
+  XCTAssertEqual(pubSub.pendingTopicUpdates.numberOfBatches, 3);
```

Comment:
Will it be difficult to validate content here to make sure archiving/unarchiving works as expected? Alternatively we may add tests for the models themself e.g. like [this one](https://github.com/firebase/firebase-ios-sdk/blob/9e5c9055fb2318071eb32fd07888af89e516df3e/FirebaseInstallations/Source/Tests/Unit/FIRInstallationsStoredAuthTokenTests.m#L29).

Reply:
Done. Yeah that would be very helpful.